### PR TITLE
Add auto-reload example

### PR DESCRIPTION
**Pull request by jplatte**
## Motivation

Auto-reload is very convenient for development, but few people are aware how easy it is to implement.

## Solution

Add an example!

### CODE REVIEW

File: examples/auto-reload/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+## Setup
+
+```sh
+cargo install cargo-watch systemfd
+```
+
+## Running
+
+```sh
+systemfd --no-pid -s http::3000 -- cargo watch -x run
```
Comment (hi-rustin) [inline examples/auto-reload/README.md:17]: Should we mention that you only can run this command under the `examples/auto-reload` dir?

**Review by hi-rustin — APPROVED** [review]
Looks good to me.

**Review by davidpdrsn — APPROVED** [review]

**Comment by matx64** [comment]
Hi @jplatte , is there any advantage between using `listenfd` compared to just `cargo-watch -c -q -w . -x "run"` ?

**Comment by jplatte** [comment]
IIRC, stopping and restarting an application that binds to a certain address / port within a small timeframe can result in the OS not allowing the new bind. Additionally, listenfd running while the Rust app is still compiling means that a client (e.g. browser) can already connect and will just wait for the server to start responding, rather than producing a cannot-connect error which often requires the user to retry explicitly once the server it up.

**Pull request merged**